File: gunhoflash/house-robber.ts
function rob(nums: number[]): number {
  let answerPrevPrev = 0;
  let answerPrev = 0;
  let answerNow = 0;

  for (const num of nums) {
    answerNow = Math.max(answerPrevPrev + num, answerPrev);
    answerPrevPrev = answerPrev;
    answerPrev = answerNow;
  }

  return Math.max(answerNow, answerPrev);
}

File: gunhoflash/house-robber.py
from typing import List

class Solution:
  def rob(self, nums: List[int]) -> int:
    answerPrevPrev = answerPrev = answerNow = 0

    for num in nums:
      answerNow = max(answerPrevPrev + num, answerPrev)
      answerPrevPrev = answerPrev
      answerPrev = answerNow

    return max(answerNow, answerPrev)
